cmd/operator-sdk/generate/csv.go

```diff
@@ -97,60 +110,42 @@ Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config
 
 func (c csvCmd) run() error {
 
-	absProjectPath := projutil.MustGetwd()
-	cfg := &input.Config{
-		AbsProjectPath: absProjectPath,
-		ProjectName:    filepath.Base(absProjectPath),
-	}
-	if projutil.IsOperatorGo() {
-		cfg.Repo = projutil.GetGoPkg()
-	}
-
 	log.Infof("Generating CSV manifest version %s", c.csvVersion)
 
-	csvCfg, err := catalog.GetCSVConfig(c.csvConfigPath)
-	if err != nil {
-		return err
-	}
 	if c.operatorName == "" {
-		// Use config operator name if not set by CLI, i.e. prefer CLI value over
-		// config value.
-		if c.operatorName = csvCfg.OperatorName; c.operatorName == "" {
-			// Default to using project name if both are empty.
-			c.operatorName = filepath.Base(absProjectPath)
-		}
+		c.operatorName = filepath.Base(projutil.MustGetwd())
 	}
-
-	s := &scaffold.Scaffold{}
-	csv := &catalog.CSV{
-		CSVVersion:     c.csvVersion,
-		FromVersion:    c.fromVersion,
-		ConfigFilePath: c.csvConfigPath,
-		OperatorName:   c.operatorName,
-	}
-	err = s.Execute(cfg, csv)
-	if err != nil {
-		return fmt.Errorf("catalog scaffold failed: %v", err)
+	cfg := gen.Config{
+		OperatorName: c.operatorName,
+		Inputs:       c.inputs,
+		OutputDir:    c.outputDir,
 	}
 
-	gcfg := gen.Config{
-		OperatorName: c.operatorName,
-		OutputDir:    filepath.Join(gencatalog.OLMCatalogDir, c.operatorName),
+	csv := gencatalog.NewCSV(cfg, c.csvVersion, c.fromVersion)
```

Comment:
I'm now thinking of using a generator type-specific `Options` struct, ex. `CSVOptions` to configure the generator constructor, and removing `Inputs` and `OutputDir` from `Config`. It would look something like:

```go
type CSVOptions struct {
	gen.Config
	OutputDir 	string
	CSVVersion  string
	FromVersion string
	DeployDir	string
	APIsDir		string
}

func NewCSV(opts CSVOptions) Generator {
	...
}
```

An options struct is idiomatic and makes it easier to deal with defaults. Setup is a little more verbose because you're not only sharing one config, with the tradeoffs of generator-specific configuration captured in one place (no constructor args + inputs map) and the nice properties of struct fields vs those of map indexing.

`OutputDir` may be different for two generators run in the same command, so it makes sense to move it out of `gen.Config` and into generator-specific options.

Using keyed inputs still makes sense; we'd move the key definitions to `cmd/operator-sdk/generate/csv.go`.

Reply:
@estroz Agreed. I'd probably go even further to remove the Config altogether if there isn't much else besides the Inputs and OutputDir.
The Inputs and OutputDir in the generator config would be better suited as defined options fields per generator rather than a generic inputs/output for the sake of sharing it between an admittedly small number of generators. And the map indexing is harder to read vs simple option fields.

However to avoid increasing the scope of this PR (and refactoring the CRD generator as well) I'd suggest using the Generator config as it is right now so we can get `generate csv` to have configurable inputs and output flags.

And I'll add a TODO (for a later PR) in the Generator Config to refactor out Inputs and OutputDir into generator specific options.

WDYT?